src/NServiceBus.Core/Routing/RoutingComponent.cs

```diff
@@ -43,16 +54,26 @@ public static RoutingComponent Initialize(Configuration configuration, Transport
             {
                 distributionPolicy.SetDistributionStrategy(distributionStrategy);
             }
-            
+
             configuration.ConfiguredUnicastRoutes?.Apply(unicastRoutingTable, conventions);
 
+            var isSendOnlyEndpoint = receiveConfiguration == null;
+            if (isSendOnlyEndpoint == false)
+            {
+                pipelineSettings.Register(new ApplyReplyToAddressBehavior(
```

Comment:
we moved all pipeline registrations from the routing behavior to the send component while routing only provides the actual routing logic. This change would break this approach again by applying routing specific behaviors to the pipeline. Not sure the routing component is the right place for this tbh.

Reply:
Talked this through with Tim and we thing this is fine since it moves all routing logic to a single place. We'll see how the evolves in terms on dependencies btw components but at least this makes the dependency more visible